### PR TITLE
dependency improvement for KNP

### PR DESCRIPTION
**Pull request by KoichiYasuoka**
Improvement for
* 解析格:ガ２ 「象は鼻が長い」
* 係:ト格 with 並列タイプ 「リンゴとナシとミカン」
* 数詞 of 名詞 「三匹の子ぶた」
* 接頭辞 「御盃を相交わす」

### CODE REVIEW

**Comment by KoichiYasuoka** [comment]
"test extras / build (juman)" failed.  Umm...

File: camphr/pipelines/knp/dependency_parser.py [hunk 1]
```diff
@@ -90,3 +99,44 @@ def _get_child_dep(tag: Token) -> str:
...
         return "clf" if pp == NUM else "flat"
+
+
+def _modify_head_flat(heads: List[Token]) -> List[Token]:
+    s = [t for t in heads]
+    for i, t in enumerate(s):
+        if not t.tag_.startswith("接頭辞"):
+            continue
+        x = [u for u in t.rights]  # type: ignore
+        h = x[0]
```
Comment (tamuhey) [inline camphr/pipelines/knp/dependency_parser.py:110]:
```suggestion
        if len(x) == 0:
            continue
        h = x[0]
```

**Comment by tamuhey** [comment]
Thanks for contributing!
Please add the followings as test cases in `test_knp_dependency_parser`
```
解析格:ガ２ 「象は鼻が長い」
係:ト格 with 並列タイプ 「リンゴとナシとミカン」
数詞 of 名詞 「三匹の子ぶた」
接頭辞 「御盃を相交わす」
```

File: camphr/pipelines/knp/dependency_parser.py (hunk 1 above)
Comment (KoichiYasuoka) [reply camphr/pipelines/knp/dependency_parser.py:112]: Oops.

**Comment by KoichiYasuoka** [comment]
I've found that JUMAN++ mistakes ナシ for 形容詞, but not JUMAN. Well, I'll change the example to バナナ for the test.

**Comment by KoichiYasuoka** [comment]
I've also found that JUMAN cuts 子ぶた into 子 and ぶた, but not JUMAN++. Umm, I'll change it for 三匹の豚.

**Comment by tamuhey** [comment]
Please merge `origin/master`

File: tests/pipelines/knp/test_dependency_parser.py [hunk 2]
```diff
@@ -26,3 +26,35 @@ def test_dependency_deps(nlp, text, deps):
...
+        assert token.head.i == headi
+        assert token.dep_ == depi
+
+
+@pytest.mark.parametrize("text,heads,deps", [("御盃を相交わす", [1, 4, 1, 4, 4], ["compound", "obj", "case", "advmod", "ROOT"])])
+def test_dependency_parse_deps_4(nlp, text, heads, deps):
+    doc = nlp(text)
+    for token, headi, depi in itertools.zip_longest(doc, heads, deps):
+        assert token.head.i == headi
+        assert token.dep_ == depi
```
Comment (tamuhey) [inline tests/pipelines/knp/test_dependency_parser.py:60]: These tests can be combined into one.

```suggestion
@pytest.mark.parametrize("text,heads,deps", [
("象は鼻が長い", [4, 0, 4, 2, 4], ["dislocated", "case", "nsubj", "case", "ROOT"]),
("リンゴとバナナとミカン", [0, 0, 0, 2, 0], ["ROOT", "case", "conj", "case", "conj"]),
("三匹の豚", [3, 0, 0, 3], ["nummod", "clf", "case", "ROOT"]),
("御盃を相交わす", [1, 4, 1, 4, 4], ["compound", "obj", "case", "advmod", "ROOT"])

])
def test_dependency_parse_deps(nlp, text, heads, deps):
    doc = nlp(text)
    for token, headi, depi in itertools.zip_longest(doc, heads, deps):
        assert token.head.i == headi
        assert token.dep_ == depi
```

**Comment by tamuhey** [comment]
Please merge `origin/master` into this branch for test.

File: tests/pipelines/knp/test_dependency_parser.py (hunk 2 above)
Comment (KoichiYasuoka) [reply tests/pipelines/knp/test_dependency_parser.py]: Thank you, and how about "太郎が本を読む" ?

**Comment by KoichiYasuoka** [comment]
I've written about [this commit](https://github.com/PKSHATechnology-Research/camphr/pull/47/commits/72161e0d60734ad2ac63eab22b9fbcbc7e7b9c9d) for `conj` in [my diary](https://srad.jp/~yasuoka/journal/638356/).

**Comment by tamuhey** [comment]
Cool! Thank you very much!

**Pull request merged**